gradio/themes/builder.py

```diff
@@ -0,0 +1,945 @@
+        )
+        attach_rerender(neutral_hue.select(load_color, neutral_hue, neutral_hues).then)
+        for hue_set in (primary_hues, secondary_hues, neutral_hues):
+            for hue in hue_set:
+                attach_rerender(hue.blur)
+
+        def load_size(size_name):
+            size = [size for size in sizes if size.name == size_name][0]
+            return [getattr(size, i) for i in size_range]
+
+        attach_rerender(text_size.change(load_size, text_size, text_sizes).then)
+        attach_rerender(
+            spacing_size.change(load_size, spacing_size, spacing_sizes).then
+        )
+        attach_rerender(radius_size.change(load_size, radius_size, radius_sizes).then)
+
+        attach_rerender(
+            load_theme_btn.click(load_theme, base_theme_dropdown, theme_inputs).then
+        )
+
+        for theme_box in (
+            text_sizes + spacing_sizes + radius_sizes + main_fonts + mono_fonts
+        ):
+            attach_rerender(theme_box.blur)
+            attach_rerender(theme_box.submit)
+        for theme_box in theme_var_input:
+            attach_rerender(theme_box.blur)
+            attach_rerender(theme_box.select)
+        for checkbox in main_is_google + mono_is_google:
+            attach_rerender(checkbox.select)
+
+        dark_mode_btn.click(
+            None,
+            None,
+            None,
+            _js="""() => {
+            if (document.querySelectorAll('.dark').length) {
+                document.querySelectorAll('.dark').forEach(el => el.classList.remove('dark'));
+            } else {
+                document.querySelector('body').classList.add('dark');
+            }
+        }""",
+        )
+
+        def undo(history_var):
+            if len(history_var) <= 1:
+                return {history: gr.skip()}
+            else:
+                history_var.pop()
+                old = history_var.pop()
+                return history_var, old[0], *old[1]
+
+        attach_rerender(
+            undo_btn.click(
+                undo, [history], [history, base_theme_dropdown] + theme_inputs
+            ).then
+        )
+
+        def upload_to_hub(data):
+            theme_url = data[current_theme].push_to_hub(
+                repo_name=data[theme_repo_name],
+                version=data[theme_version],
+                hf_token=data[theme_hf_token],
+                theme_name=data[theme_name],
+                description=data[theme_description],
+            )
+            return gr.Markdown.update(
+                value=f"Theme uploaded [here!]({theme_url})!", visible=True
+            )
+
+        upload_to_hub_btn.click(
+            upload_to_hub,
+            {
+                current_theme,
+                theme_name,
+                theme_hf_token,
+                theme_repo_name,
+                theme_version,
+                theme_description,
+            },
+            theme_upload_status,
+        )
+
+
+if __name__ == "__main__":
+    demo.launch()
```

Comment:
Suggestion:
```suggestion
    demo.launch(inline=False, inbrowser=True)
```

Reply:
Well this actually isn't where it's launched really, it's launched from  `themes/__init__.py` as:
```python
def builder(*args, **kwargs):
    from gradio.themes.builder import demo

    demo.launch(*args, **kwargs)
```
So users can pass any launch args